packages/router/src/history.ts

```diff
@@ -32,9 +36,35 @@ function preserveThemePreview( params: Record< string, any > ) {
 	return { ...params, wp_theme_preview: currentThemePreview };
 }
 
-function push( params: Record< string, any >, state: Record< string, any > ) {
-	const search = buildQueryString( preserveThemePreview( params ) );
-	return originalHistoryPush.call( history, { search }, state );
+function push(
+	params: Record< string, any >,
+	state: Record< string, any >,
+	options: PushOptions = {}
```

Comment:
This change basically adds support for view transitions to our "private" router.